### PR TITLE
Adds Explosive Handcuffs as a Security-only Traitor item

### PR DESCRIPTION
**Pull request by ghost**
This is one of the two Security-only Traitor items I have planned, took long enough and caused enough modifications to get a first PR
- Adds Syndicate Handcuffs (Stylized Syndicuffs). Those look like handcuffs on every aspect, but they contain a C4 charge and are filled with electronics. They cost 2 TC for every unit
- The handcuffs have two modes, they either explode three seconds after being engaged (someone is handcuffed) or immediately when disengaged (handcuffs are removed for any reason). Modes are toggled by pulling the rotating arm all the way back (using the handcuffs)
- Explosion is a (0, 1, 3). This will crit the target and might breach the hull, and will "lightly" (will generally lead to internal bleeding and fractures) damage anyone close by
- Handcuffs can be triggered prematurely by explosions or EMP blasts (might not work when in a mob's inventory due to spaghetti code as far as explosions go, EMPs should be fine as demonstrated by headsets and such)
- Added new procs to handle handcuffs application, removal and deletion. This was done to have the required procs actually work without shitcoding in the inventory file, but could be of use later
- Added a sound var to remove a few lines of shitcode, also allows custom sounds for your top-tier adminbus cuffs
- Fixed handcuffing message (used to be 'User is trying to put a handcuffs on Shitler'), used a plural check so that at least this one is fixed
- Updated a bunch of things related to those changes

Explosive Handcuffs cost as much as C4 (2 TC). The balance reasoning is that Explosive Handcuffs can be easily sneaked in on a -somewhat- innocent Security Officer, are not obvious to onlookers until it blows up, can be applied without causing alarm and can cause slightly more serious damage around the target, BUT it does not gib the target, meaning he can easily be dragged off to be cloned unlike the block of C4 which will literally remove the target outside of its head, which can be picked up and binned, or cut open and used for a braincake.

### CODE REVIEW

**Comment by DenkouZ** [comment]
We need more job exclusive traitor items to spur people into being more creative with their jobs besides stealing the spare and starting a plama flood in an atmos suit with the armory in their bag. Nice work. Inventive and devious.

File: code/game/objects/items/weapons/handcuffs.dm [hunk 1]
```diff
@@ -64,48 +67,145 @@
...
-				spawn( 0 )
-					if(istype(src, /obj/item/weapon/handcuffs/cable))
-						playsound(get_turf(src), 'sound/weapons/cablecuff.ogg', 30, 1, -2)
-					else
-						playsound(get_turf(src), 'sound/weapons/handcuffs.ogg', 30, 1, -2)
-					O.process()
+				handcuffs_apply(C, user)
 			return
+
+/obj/item/weapon/handcuffs/proc/detonate(var/countdown = 1)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/handcuffs.dm:74]: That's not how OOP works you absolute dork.

File: code/game/objects/items/weapons/handcuffs.dm [hunk 2]
```diff
@@ -64,48 +67,145 @@
...
+		O.place = "handcuff"
+		C.requests += O
+		spawn()
+			if(istype(src, /obj/item/weapon/handcuffs/cable))
+				feedback_add_details("handcuffs", "C")
+			else
+				feedback_add_details("handcuffs", "H")
+			playsound(get_turf(src), cuffing_sound, 30, 1, -2)
+			O.process()
+			if(mode == SYNDICUFFS_ON_APPLY && !charge_detonated)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/handcuffs.dm]: Really?

File: code/game/objects/items/weapons/handcuffs.dm [hunk 3]
```diff
@@ -64,48 +67,145 @@
...
+		O.target = C
+		O.item = user.get_active_hand()
+		O.s_loc = user.loc
+		O.t_loc = C.loc
+		O.place = "handcuff"
+		C.requests += O
+		spawn()
+			playsound(get_turf(src), cuffing_sound, 30, 1, -2)
+			O.process()
+			if(mode == SYNDICUFFS_ON_APPLY && !charge_detonated)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/handcuffs.dm]: "Oh this is simple no way dylan can fuck this up"

File: code/game/objects/items/weapons/handcuffs.dm [hunk 4]
```diff
@@ -64,48 +67,145 @@
...
+				detonate(1)
+
+/obj/item/weapon/handcuffs/proc/handcuffs_remove(var/mob/living/carbon/C)
+
+	if(mode == SYNDICUFFS_ON_REMOVE && !charge_detonated)
```
Comment (PJB3005) [inline code/game/objects/items/weapons/handcuffs.dm]: https://en.wikipedia.org/wiki/Object-oriented_programming

File: code/game/objects/items/weapons/handcuffs.dm [hunk 5]
```diff
@@ -64,48 +67,145 @@
...
+
+	mode = SYNDICUFFS_ON_APPLY
+	var/countdown_time = 30
+
+/obj/item/weapon/handcuffs/syndicate/attack_self(mob/user)
+
+	switch(mode)
+		if(SYNDICUFFS_ON_REMOVE)
+			mode = SYNDICUFFS_ON_APPLY
+			user << "<span class='notice'>You pull the rotating arm back until you hear two clicks. \The [src] will detonate a few seconds after being applied.</span>"
```
Comment (PJB3005) [inline code/game/objects/items/weapons/handcuffs.dm]: `to_chat()`

File: code/game/objects/items/weapons/handcuffs.dm [hunk 6]
```diff
@@ -64,48 +67,145 @@
...
+		if(SYNDICUFFS_ON_APPLY)
+			mode = SYNDICUFFS_ON_REMOVE
+			user << "<span class='notice'>You pull the rotating arm back until you hear one click. \The [src] will detonate when removed.</span>"
```
Comment (PJB3005) [inline code/game/objects/items/weapons/handcuffs.dm]: `to_chat()`

File: code/game/objects/items/weapons/handcuffs.dm [hunk 7]
```diff
@@ -64,48 +67,145 @@
...
+	else
+		explosion(get_turf(src), 0, 1, 3, 0)
+		qdel(src)
+
+/obj/item/weapon/handcuffs/Destroy()
+
+	if(iscarbon(loc)) //Inventory shit
+		var/mob/living/carbon/C = loc
+		if(C.handcuffed)
+			C.handcuffed.loc = C.loc //Standby while we delete this shit
```
Comment (PJB3005) [inline code/game/objects/items/weapons/handcuffs.dm:204]: Just nullspace it?

**Comment by Imagebard** [comment]
Holy fuck PJB

File: code/game/objects/items/weapons/handcuffs.dm [hunk 8]
```diff
@@ -15,45 +19,44 @@
...
 	if(istype(src, /obj/item/weapon/handcuffs/cyborg) && isrobot(user))
 		if(!C.handcuffed)
 			var/turf/p_loc = user.loc
 			var/turf/p_loc_m = C.loc
-			playsound(get_turf(src), 'sound/weapons/handcuffs.ogg', 30, 1, -2)
-			for(var/mob/O in viewers(user, null))
-				O.show_message("<span class='danger'>[user] is trying to put handcuffs on [C]!</span>", 1)
+			playsound(get_turf(src), cuffing_sound, 30, 1, -2)
+			user.visible_message("<span class='danger'>[user] is trying to handcuff \the [C]!</span>", \
+								 "<span class='danger'>You try to handcuff on \the [C]!</span>")
```
Comment (Intigracy) [inline code/game/objects/items/weapons/handcuffs.dm]: Nice grammar fam.

**Comment by ghost** [comment]
PJBreadsyourPR.gif

**Comment by Intigracy** [comment]
tbh PJB's right he should have just overrode the parent procs instead of changing them so much.

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 9]
```diff
@@ -536,7 +535,7 @@
 		if(isrobot(source) && place != "handcuff")
 			qdel(src)
 		for(var/mob/O in viewers(target, null))
-			O.show_message("<span class='danger'>[source] is trying to put \a [item] on [target]</span>", 1)
+			O.show_message("<span class='danger'>[source] is trying to put [item.gender == PLURAL ? "[item]" : "\a [item]"] on [target]</span>", 1)
```
Comment (Intigracy) [inline code/modules/mob/living/carbon/human/inventory.dm]: "\the [item]" : "\a [item]"

Comment (PJB3005) [inline code/modules/mob/living/carbon/human/inventory.dm]: Yeah I never see a use for "a" in SS13.

Comment (Intigracy) [inline code/modules/mob/living/carbon/human/inventory.dm]: or both the, idc, but the first one should have a the

File: code/game/objects/items/weapons/handcuffs.dm (hunk 7 above)
Comment (ghost) [inline code/game/objects/items/weapons/handcuffs.dm:204]: Handled by the proc below, this moves the thing out of the mob's inventory so inventory removal works properly

**Comment by ghost** [comment]
The problem with the parent procs is that inventory procs handle it, which would be a terrible place to put handcuff detonation procs in, and caused me a lot of undue problems

I'd rather have the handcuffs handle their own explosion

**Comment by gbasood** [comment]
You can handle the explosion procs within those procs just fine.

**Comment by ghost** [comment]
And I can handle it fine this way too

OOP goes both ways, in that case the inventory still handles the nitty gritty stuff, and the handcuffs handle calling procs without adding extra code to the already massive inventory.dm file

**Comment by PJB3005** [comment]
Dylan what the fuck are you on about handle the exploding in an override to the handcuffing procs on the explosive subtype you goddammit lunatic.

File: code/game/objects/items/weapons/handcuffs.dm (hunk 1 above)
Comment (Intigracy) [inline code/game/objects/items/weapons/handcuffs.dm:74]: it's literally just adding handcuffs/syndicate right here.

Comment (PJB3005) [inline code/game/objects/items/weapons/handcuffs.dm:74]: Why is the detonate proc on the main handcuff type.

Comment (ghost) [inline code/game/objects/items/weapons/handcuffs.dm:74]: Avoids type checks

Comment (Intigracy) [inline code/game/objects/items/weapons/handcuffs.dm:74]: how is if(mode == SYNDICUFFS_ON_APPLY && !charge_detonated) any different from a type check?

**Comment by Brolaire** [comment]
when

File: code/modules/mob/living/carbon/human/inventory.dm [hunk 10]
```diff
@@ -536,7 +535,7 @@
 		if(isrobot(source) && place != "handcuff")
 			qdel(src)
 		for(var/mob/O in viewers(target, null))
-			O.show_message("<span class='danger'>[source] is trying to put \a [item] on [target]</span>", 1)
+			O.show_message("<span class='danger'>[source] is trying to put [item.gender == PLURAL ? "\the [item]" : "\a [item]"] on [target]</span>", 1)
```
Comment (clusterfack) [inline code/modules/mob/living/carbon/human/inventory.dm]: Oh jesus christ use /s

Comment (ghost) [inline code/modules/mob/living/carbon/human/inventory.dm:538]: Que ?

Comment (Intigracy) [inline code/modules/mob/living/carbon/human/inventory.dm:538]: [source] is trying to put \the [item]\s on [target]

**Comment by ghost** [comment]
Welp

**Pull request closed without merging**